crates/wkg/src/main.rs

```diff
@@ -95,6 +98,53 @@ enum Commands {
     Wit(WitCommands),
 }
 
+#[derive(Args, Debug)]
+struct ConfigArgs {
+    /// The default registry domain to use. Overrides configuration file(s).
+    #[arg(long = "default", value_name = "DEFAULT")]
+    default: Option<Registry>,
+
+    #[command(flatten)]
+    common: Common,
+}
+
+impl ConfigArgs {
+    pub async fn run(self) -> anyhow::Result<()> {
+        // use config path provided, otherwise global config path
+        let path = if let Some(path) = self.common.config {
+            path
+        } else {
+            Config::global_config_path()
+                .ok_or(anyhow::anyhow!("global config path not available"))?
+        };
+
+        // read file or use default config (not empty config)
+        let mut config = match tokio::fs::read_to_string(&path).await {
+            Ok(contents) => Config::from_toml(&contents)?,
+            Err(err) if err.kind() == std::io::ErrorKind::NotFound => Config::default(),
+            Err(err) => return Err(anyhow::anyhow!("error reading config file: {0}", err)),
+        };
+
+        if let Some(default) = self.default {
+            // set default registry
+            config.set_default_registry(Some(default));
+
+            // write config file
+            config.to_file(&path).await?;
```

Comment:
Doesn't need to happen immediately but it would be nice to eventually use https://docs.rs/toml_edit/0.22.22/toml_edit/ to update config without blowing away user formatting, comments, etc.

Reply:
A bit bigger of a refactor. Prefer in a follow on PR.

Reply:
Created https://github.com/bytecodealliance/wasm-pkg-tools/issues/101